tcms/testplans/tests/test_new_plan.py

```diff
@@ -96,16 +126,87 @@ def _test_plan_create_new(self, is_active):
+class TestNewTestPlanViewPermissions(PermissionsTestCase):
+    add_testplan_permission = "testplans.add_testplan"
+    permission_label = add_testplan_permission
+    http_method_names = ["get", "post"]
+    url = reverse("plans-new")
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.post_data = {"description": "test"}
+        super().setUpTestData()
+
+        cls.plan_name = 'plan_name'
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.plan_type = PlanTypeFactory()
+        cls.test_plan = TestPlanFactory()
+
+        user_should_have_perm(cls.tester, perm="testplans.view_testplan")
+
+        cls.post_data.update(
+            {
+                "author": cls.tester.pk,
+                "product": cls.product.pk,
+                "product_version": cls.product_version.pk,
+                "type": cls.plan_type.pk,
+                "name": cls.plan_name,
+                "email_settings-0-auto_to_plan_author": "on",
+                "email_settings-0-auto_to_case_owner": "on",
+                "email_settings-0-auto_to_case_default_tester": "on",
+                "email_settings-0-notify_on_case_update": "on",
+                "email_settings-0-notify_on_plan_update": "on",
+                "email_settings-0-id": cls.test_plan.emailing.pk,
+                "email_settings-TOTAL_FORMS": "1",
+                "email_settings-INITIAL_FORMS": "1",
+                "email_settings-MIN_NUM_FORMS": "0",
+                "email_settings-MAX_NUM_FORMS": "1",
+                "is_active": True,
+            }
+        )
 
-        self.assertRedirects(response, reverse('tcms-login') + '?next=' + self.location)
+    def verify_get_with_permission(self):
+        response = self.client.get(self.url)
 
-    def test_post_with_no_perm_redirects_to_login(self):
-        remove_perm_from_user(self.user, self.add_testplan_permission)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, _('Create new TestPlan'))
+
+    def verify_post_with_permission(self):
+        response = self.client.post(self.url, self.post_data, follow=True)
+        test_plan = TestPlan.objects.get(
+            name=self.post_data["name"],
+            product=self.post_data["product"],
+        )
+        redirect_url = reverse('test_plan_url', args=[test_plan.pk, slugify(test_plan.name)])
 
-        response = self.client.post(self.location, self.request, follow=True)
+        self.assertRedirects(response, redirect_url)
+        self._assert_test_plan(test_plan)
 
-        self.assertRedirects(response, reverse('tcms-login') + '?next=' + self.location)
+    def _assert_test_plan(self, test_plan):
+        self.assertEqual(test_plan.author, self.tester)
+        self.assertEqual(test_plan.product, self.product)
+        self.assertEqual(test_plan.product_version, self.product_version)
+        self.assertEqual(test_plan.type, self.plan_type)
```

Comment:
All of these are incorrect b/c you compare the record from the DB with the values which were passed via POST and if the POST request returned a success these will always be the same.

The important behavior is to verify that after the redirect we've actually landed on the page showing the newly created object. assertRedirects above does a good job of it. assertContains on a unique plan name would compliment that and it would be enough. No need to check all fields.

Reply:
I've made the changes, pinging because I can't re-request a review @atodorov.